pkg/cmd/template/cmd.go

```diff
@@ -150,6 +150,11 @@ func (o *Options) RunWithFiles(in Input, ui ui.UI) Output {
 		return Output{Err: err}
 	}
 
+	err = rootLibraryExecution.ValidateValues(values)
```

Comment:
After taking a step back and reevaluating the change set, I think that it would make sense for `Values()` to return a valid set. I think I made `ValidateValues()` separate at the beginning of the implementation to get some distance from the Data Values Pre-Processing that we know and love.

Reply:
Makes sense to me... and reads like a great way to start in with stuff... I think it's generally easier to integrate in than extract out... so. 👍🏻